### PR TITLE
New `i18n_release_pull_and_compile` paver command + updated translations

### PR DESCRIPTION
**Pull request by OmarIthawi**
So, this one definitely needs a staging deployment first. I'm still not sure if it'll work well.

### Summary
 - Make the `$ paver i18n_robot_pull` work for our purposes by pull the release files.
 - This also brings back the now-reverted #380 in a better way.

### CODE REVIEW

**Comment by OmarIthawi** [comment]
[(2) Hawthorn testing: do we have the latest language packs installed?](https://trello.com/c/uNKhRxTW/4043-2-hawthorn-testing-do-we-have-the-latest-language-packs-installed)

**Comment by OmarIthawi** [comment]
[Update French Canadian language files on Tahoe](https://trello.com/c/dNMcbDQl/4143-update-french-canadian-language-files-on-tahoe)

File: pavelib/i18n.py [hunk 1]
```diff
@@ -280,6 +283,47 @@ def i18n_release_pull():
...
+    """
+    Pull release-specific translations from Transifex, generate po and mo files, and validate.
+
+    Appsembler: This is what we use for Tahoe.
+    """
+    settings = options.get('settings', DEFAULT_SETTINGS)
+
+    sh('git clean -fdX conf/locale/rtl')
+    sh('git clean -fdX conf/locale/eo')
+    print "\n\nValidating translations with `i18n_tool validate`..."
```
Comment (thraxil) [inline pavelib/i18n.py]:
```suggestion
    print("\n\nValidating translations with `i18n_tool validate`...")
```

**Comment by OmarIthawi** [comment]
@melvinsoft @thraxil this addresses the issues in #395 and #380 . Please take another look.

File: pavelib/i18n.py [hunk 2]
```diff
@@ -280,6 +282,51 @@ def i18n_release_pull():
...
```
Comment (thraxil) [inline pavelib/i18n.py]:
```suggestion
    print("\n\nValidating translations with `i18n_tool validate`...")
```

Comment (OmarIthawi) [reply pavelib/i18n.py]: Thanks @thraxil!

**Pull request merged**